File: Assets/scripts/Globals.cs
using UnityEngine;

public class Globals : MonoBehaviour 
{
	public static Globals GM;

	public static bool sound = true;

	public static int score = 0;
	public static int goldenStool = 0;

	public static int highScore = 0;

	public static int totalBugKills = 0;
	public static int totalGamesPlayed = 0;
	public static int totalShotsFired = 0;
	public static int totalHits = 0;

	public static int tempGameBugsKilled = 0;
	public static int tempShotsFired = 0;

	void Awake()
	{

		if(GM != null)
			GameObject.Destroy(GM);
		else
			GM = this;
		DontDestroyOnLoad(this);
	}
}

File: vite/src/game/Globals.ts
/**
 * Global game state (port of Unity Globals.cs).
 */

export const Globals = {
  sound: true,
  lowPowerMode: false,
  antialias: true,
  score: 0,
  goldenStool: 0,
  highScore: 0,
  totalBugKills: 0,
  totalGamesPlayed: 0,
  totalShotsFired: 0,
  totalHits: 0,
  tempGameBugsKilled: 0,
  tempShotsFired: 0,
  inGame: false,
  selectedLevel: 'gasStation' as 'gasStation' | 'portaPotty',
};

export function resetForNewGame(): void {
  Globals.score = 0;
  Globals.tempGameBugsKilled = 0;
  Globals.tempShotsFired = 0;
  Globals.inGame = true;
}

export function commitGameOverStats(): void {
  Globals.totalBugKills += Globals.tempGameBugsKilled;
  Globals.totalGamesPlayed += 1;
  Globals.totalShotsFired += Globals.tempShotsFired;
  Globals.totalHits += Globals.tempGameBugsKilled;
}
